Sources/UI/SwiftUI/View Modifiers/PinchToZoomModifier.swift

```diff
@@ -0,0 +1,164 @@
+//
+//  PinchToZoomModifier.swift
+//  PovioKit
+//
+//  Created by Yll Fejziu on 13/05/2024.
+//  Copyright © 2024 Povio Inc. All rights reserved.
+//
+
+import SwiftUI
+
+/// https://stackoverflow.com/a/59878898
+public struct PinchToZoomModifier: ViewModifier {
+  @State var scale: CGFloat = 1.0
+  @State var anchor: UnitPoint = .center
+  @State var offset: CGSize = .zero
+  @State var isPinching: Bool = false
+
+  public func body(content: Content) -> some View {
+    content
+      .scaleEffect(scale, anchor: anchor)
+      .offset(offset)
+      .animation(isPinching ? .none : .spring(), value: scale)
+      .overlay(
+        PinchZoom(
+          scale: $scale,
+          anchor: $anchor,
+          offset: $offset,
+          isPinching: $isPinching
+        )
+      )
+  }
+}
+
+public extension View {
+  func pinchToZoom() -> some View {
+    self.modifier(PinchToZoomModifier())
+  }
+}
+
+// MARK: - Private
+private protocol PinchZoomViewDelgate: AnyObject {
+  func pinchZoomView(_ pinchZoomView: PinchZoomView, didChangePinching isPinching: Bool)
+  func pinchZoomView(_ pinchZoomView: PinchZoomView, didChangeScale scale: CGFloat)
+  func pinchZoomView(_ pinchZoomView: PinchZoomView, didChangeAnchor anchor: UnitPoint)
+  func pinchZoomView(_ pinchZoomView: PinchZoomView, didChangeOffset offset: CGSize)
+}
+
+private class PinchZoomView: UIView {
+  weak var delegate: PinchZoomViewDelgate?
+
+  private(set) var scale: CGFloat = 0 {
+    didSet {
+      delegate?.pinchZoomView(self, didChangeScale: scale)
+    }
+  }
+
+  private(set) var anchor: UnitPoint = .center {
+    didSet {
+      delegate?.pinchZoomView(self, didChangeAnchor: anchor)
+    }
+  }
+
+  private(set) var offset: CGSize = .zero {
+    didSet {
+      delegate?.pinchZoomView(self, didChangeOffset: offset)
+    }
+  }
+
+  private(set) var isPinching: Bool = false {
+    didSet {
+      delegate?.pinchZoomView(self, didChangePinching: isPinching)
+    }
+  }
```

Comment:
I would avoid the delegate pattern and modify the view's state directly instead.

Reply:
We’re using UIViewRepresentable here, and if I understand correctly, your suggestion is to bind the UIKit view state directly to SwiftUI state. 
One way to achieve this would be using Combine, as it allows us to directly bind state changes without a delegate pattern. Is this what you had in mind, or did you have another approach in mind for avoiding the delegate pattern?

Reply:
No need to use Combine. One solution is to move the state into a view model and inject a reference to the view model into `PinchZoomView`.